src/dbos-executor.ts

```diff
@@ -668,6 +753,35 @@ export class DBOSExecutor implements DBOSExecutorContext {
     }, ...args)).getResult();
   }
 
+  async procedure<R>(proc: StoredProcedure<R>, params: WorkflowParams, ...args: unknown[]): Promise<R> {
+    // Create a workflow and call procedure.
+    const temp_workflow = async (ctxt: WorkflowContext, ...args: unknown[]) => {
+      const ctxtImpl = ctxt as WorkflowContextImpl;
+      return await ctxtImpl.procedure(proc, ...args);
+    };
+    return (await this.workflow(temp_workflow, { ...params, tempWfType: TempWorkflowType.procedure, tempWfName: proc.name }, ...args)).getResult();
```

Comment:
Also need to pass in `tempWfClass` so we can correctly record the class name information.